### PR TITLE
[SMALLFIX] fixing miredot type annotations

### PR DESCRIPTION
**Pull request by jsimsa**
This PR fixes miredot type annotations, which are needed to display REST API documentation properly.

### CODE REVIEW

**Comment by calvinjia** [comment]
LGTM, is there a way to enforce this (like checkstyle)?

**Comment by jsimsa** [comment]
I am not aware of one. This shows up as a "critical error"

**Comment by jsimsa** [comment]
done

**Comment by jsimsa** [comment]
@calvinjia  PTAL

**Comment by jsimsa** [comment]
Jenkins, test this please

**Comment by jsimsa** [comment]
@calvinjia not sure why the enabling the fail-on-error option fails the build on Jenkins ... it works fine on my local machine with both Java 7 and Java 8

File: core/server/intro.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    (the "License"). You may not use this work except in compliance with the License, which is
+    available at www.apache.org/licenses/LICENSE-2.0
+
+    This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+    either express or implied, as more fully set forth in the License.
+
+    See the NOTICE file distributed with this work for information regarding copyright ownership.
+
+-->
+-->
```
Comment (gpang) [inline core/server/intro.html]: are both of these 'end comment' lines required?

File: core/server/pom.xml [hunk 2]
```diff
@@ -210,6 +210,12 @@
           <license>
             cHJvamVjdHxvcmcuYWxsdXhpby5hbGx1eGlvLWNvcmUtc2VydmVyfDIwMTgtMDQtMjh8dHJ1ZXwtMSNNQ3dDRkFFM0JGSnAwZ1FRNFFaNDd5SnlRd0x3ZGhmQUFoUU1ZMkJpWWVOamhOZndvQmU3UldMRVFzZnNGUT09
           </license>
+          <output>
+            <html>
+              <intro>${project.basedir}/intro.html</intro>
```
Comment (gpang) [inline core/server/pom.xml]: should this file be in some resources/ folder? It seems a little strange to have it at the top level like this.

**Comment by gpang** [comment]
Thanks @jsimsa ! I left a few questions.

**Comment by jsimsa** [comment]
@gpang thanks, updated

**Comment by apc999** [comment]
LGTM

**Comment by gpang** [comment]
LGTM

**Pull request merged**